app/services/foreman_salt/report_importer.rb

```diff
@@ -161,9 +161,9 @@ def process_failures
       status = ConfigReportStatusCalculator.new(counters: { 'failed' => @raw.size }).calculate
       @report = ConfigReport.create(host: @host, reported_at: Time.zone.now, status: status, metrics: {})
 
-      source = Source.find_or_create('Salt')
+      source = Source.find_or_create_by(value: 'Salt')
       @raw.each do |failure|
-        message = Message.find_or_create(failure)
+        message = Message.find_or_create(value: failure)
```

Comment:
:facepalm: should have been `find_or_create_by`

Reply:
🤦🏼 sorry about that, i'll push the fix directly to master